addons/mail/static/src/views/form/form_compiler.js

```diff
@@ -78,19 +78,21 @@ export class MailFormCompiler extends ViewCompiler {
 
     compile(node, params) {
         const res = super.compile(node, params).children[0];
-        const chatterContainerHookXml = res.querySelector(".o_FormRenderer_chatterContainer");
-        if (chatterContainerHookXml) {
-            setAttributes(chatterContainerHookXml, {
-                "t-if": `!hasAttachmentViewer() and uiService.size >= ${SIZES.XXL}`,
-                "t-attf-class": `{{ uiService.size >= ${SIZES.XXL} ? "o-aside" : "" }}`,
-            });
-        }
         const attachmentViewHookXml = res.querySelector(".o_attachment_preview");
         if (attachmentViewHookXml) {
             setAttributes(attachmentViewHookXml, {
                 "t-if": `hasAttachmentViewer() and uiService.size >= ${SIZES.XXL}`,
             });
         }
+        const chatterContainerHookXml = res.querySelector(".o_FormRenderer_chatterContainer");
+        if (chatterContainerHookXml) {
+            setAttributes(chatterContainerHookXml, {
+                "t-if": attachmentViewHookXml
```

Comment:
Hello, thank you for the PR.

The correct fix should be to fix the value of `hasAttachmentViewer` instead, otherwise if its unreliable the other places where it is used will lead to bugs as well.

If the value of `hasAttachmentViewerInArch` is reliable in form controller, I believe simply using this value inside `hasAttachmentViewer` should fix the issue.

Reply:
I have not followed the recent PR that made changes in this file, what is the difference between `export class MailFormCompiler extends ViewCompiler {` and `patch(FormCompiler.prototype, 'mail', {` ?

They seem to be doing different things, but I don't really understand.

Reply:
Why is there a duplicate `hasAttachmentViewer` in form controller and in form renderer?